### DISCUSSION TITLE
Wildcard won't group results in the same folder?

### DISCUSSION THREAD

**Discussion by joaociocca** [pls-rs/pls#81] [Help Q&A] [answered]
I loved this project when I saw it in a video a friend passed to me, thought I'd ditch ls completely!

But then... I use a lot of `ls -la **.json`...

![image](https://github.com/dhruvkb/pls/assets/6473725/c2d44ae9-481c-4da5-a8c0-3dadc8a52ae3)

and `pls -d` doesn't seem to like wildcards much? I even had an alias ready for it in `pla` hehe...

![image](https://github.com/dhruvkb/pls/assets/6473725/31a73e2d-2953-445c-938b-b2856a49bc82)

**Comment by dhruvkb** [comment]
This is possible in `pls` using [name filters](https://dhruvkb.github.io/pls/features/name_filter/) `--exclude` and `--only`.

```console
$ pls --det std --only '.json'
```

**Comment by joaociocca** [comment]
Hmm... So a completely different approach to... 🤔

Em dom., 27 de ago. de 2023 08:21, Dhruv Bhanushali <
***@***.***> escreveu:

**Reply by dhruvkb** [reply]
The main reason why `pls` does this was consistency between the output for files and folders.

<img width="707" alt="Screenshot 2023-08-28 at 6 13 04 AM" src="https://github.com/dhruvkb/pls/assets/16580576/f856efc7-4671-4a17-9e6a-0be53a95ba9d">

But upon careful consideration, I agree with your points. Separating each entry wastes a lot of space and name-based filters are inadequate. So I've opened #82 to fix this. Thanks for reporting!

**Reply by joaociocca** [reply]
I'm glad my idea was of help to improve such a beautiful project!

**Comment by dhruvkb** [comment, marked as answer]
Closing as this has been recognised as an actual issue #82.
